pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -91,7 +91,8 @@ type Controller struct {
 
 	// Env is set by server to point to the environment, to allow the controller to
 	// use env data and push status. It may be null in tests.
-	Env *model.Environment
+	Env      *model.Environment
+	StopChan chan struct{}
```

Comment:
Why are these fields added here?

Reply:
As discussed the Env wasn't added just the diff thinks there was some white space changes.

Reply:
stop channel shouldn't be added here, IMO.

Reply:
I agree. This doesn't seem clean. I am not the best person to comment on go package organization, but even otherwise, this seems a bit convoluted, since stop chan is used by server, and not by the controller.